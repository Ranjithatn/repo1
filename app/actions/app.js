import {
  UPDATE_SERVER_STATUS,
  REQUEST_NFIQ_QUALITY_THRESHOLD,
  REQUEST_NFIQ_QUALITY_THRESHOLD_SUCCESS,
} from "./actionTypes";

export const requestUpdateServerStatus = (data) => {
	return {
	  type: UPDATE_SERVER_STATUS,
	  payload: data
	};
}

export const requestNFIQQualityThreshold = (data) => {
	return {
	  type: REQUEST_NFIQ_QUALITY_THRESHOLD,
	  payload: data
	};
}

export const requestUpdateNFIQQualityThreshold = (data) => {
	return {
	  type: REQUEST_NFIQ_QUALITY_THRESHOLD_SUCCESS,
	  payload: data
	};
}

