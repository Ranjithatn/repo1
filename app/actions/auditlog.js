// requestShowAuditLog
import {
  REQUEST_SHOW_AUDITLOG,
  REQUEST_SHOW_AUDITLOG_SUCCESS,
  REQUEST_AUDITLOG_PAGES,
  REQUEST_JOB_AUDIT_LOGS,
  REQUEST_JOB_AUDIT_LOGS_SUCESS,
  REQUEST_SET_AUDIT_LOGS_FILTER,
  REQUEST_RESET_AUDIT_LOGS_FILTER,
  REQUEST_AUDIT_LOG_SORT,
  REQUEST_SET_AUDIT_LOG_SORT,
  REQUEST_RESET_AUDIT_LOG_SORT
} from "./actionTypes";

export const requestShowAuditLog = (id, status, startDate, endDate, Pages) => {
  const data = {
    jobID: id,
    jobStatus: status,
    jobStartDate: startDate,
    jobEndDate: endDate,
    PageDetails: Pages
  };

  return {
    type: REQUEST_SHOW_AUDITLOG,
    payload: data
  };
};
export const requestOpenAudiLogSuccess = AuditlogData => ({
  type: REQUEST_SHOW_AUDITLOG_SUCCESS,
  payload: AuditlogData
});

type pageSettings = {
  pageNo: number | string
};

// export const requestAuditLogPages = (payload: ?pageSettings) => {
//   type: REQUEST_AUDITLOG_PAGES;
// };

export const requestJobAuditLogs = JobID => ({
  type: REQUEST_JOB_AUDIT_LOGS,
  payload: JobID
});

export const requestJobAuditLogsSucess = LogDetails => ({
  type: REQUEST_JOB_AUDIT_LOGS_SUCESS,
  payload: LogDetails
});
export const requestResetAditlogFilter = () => ({
  type: REQUEST_RESET_AUDIT_LOGS_FILTER
});

export const requestSetAuditLogFilter = (id, status, startDate, endDate, Pages) => {
  const data = {
    jobID: id,
    jobStatus: status,
    jobStartDate: startDate,
    jobEndDate: endDate,
    PageDetails: Pages
  };

  return {
    type: REQUEST_SET_AUDIT_LOGS_FILTER,
    payload: data
  };
};
export const requrstAuditLogSort = (column, type,page) => {
  const data = {
    column: column,
    type: type,
    page:page
  
  };

  return {
    type: REQUEST_AUDIT_LOG_SORT,
    payload: data
  };
};
export const requestSetAuditlogSort = (column, type,) => {
  const data = {
    column: column,
    type: type
  
  };

  return {
    type: REQUEST_SET_AUDIT_LOG_SORT,
    payload: data
  };
};
export const requestResetAuditlogSort = () => ({
 
    type: REQUEST_RESET_AUDIT_LOG_SORT
 
});
