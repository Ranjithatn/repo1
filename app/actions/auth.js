// @flow
import {
  REQUEST_LOGIN,
  REQUEST_LOGIN_SUCCESS,
  REQUEST_LOGIN_FAILED,
  REQUEST_LOGOUT,
  REQUEST_LOGOUT_SUCCESS,
  REQUEST_SET_NEW_UPDATE,
  REQUEST_SHOW_UPDATE_MESSAGE,
  REQUEST_SET_DOWNLOAD,
  REQUEST_SET_STATUS,
  REQUEST_CHECK_FOR_APP_UPDATE,
  REQUEST_LOGIN_POPUP,
  REQUEST_SET_LOOKUPS,
  REQUEST_SET_LOOKUP_CRIME_TYPES,
  REQUEST_SET_LOOKUP_NATIONALITY,
  REQUEST_SET_USER_INFO,
} from "./actionTypes";

type credentials = {
  username?: string,
  password?: string
};

export const requestLogin = (payload: credentials, history) => {
  return {
    type: REQUEST_LOGIN,
    payload,
    history
  };
};


export const requestLoginPopup = (payload: credentials) => {
  return {
    type: REQUEST_LOGIN_POPUP,
    payload,
  };
};


export const requestLoginSuccess = (authData: {
  jwt: string,
  roles: any,
  parsedRoles: array,
}) => {
  return {
    type: REQUEST_LOGIN_SUCCESS,
    payload: authData
  };
};

export const requestLoginFailed = (error: string) => {
  return {
    type: REQUEST_LOGIN_FAILED,
    payload: error
  };
};

export const requestLogout = history => {
  return {
    type: REQUEST_LOGOUT,
    history
  };
};

export const requestLogoutSuccess = () => {
  return {
    type: REQUEST_LOGOUT_SUCCESS
  };
};
export const requestSetNewUpdate = () => {
  return {
    type: REQUEST_SET_NEW_UPDATE
  };
};
export const requestSetDownload = version => {
  return {
    type: REQUEST_SET_DOWNLOAD,
    payload: version
  };
};
export const requestShowUpdateMessage = msg => {
  return {
    type: REQUEST_SHOW_UPDATE_MESSAGE,
    payload: msg
  };
};
export const requestSetStatus = status => {
  return {
    type: REQUEST_SET_STATUS,
    payload: status
  };
};



export const requestCheckForAppUpdate = () => {
  return {
    type: REQUEST_CHECK_FOR_APP_UPDATE,
  };
};

export const requestSetLookups = (data) => {
  return {
    type: REQUEST_SET_LOOKUPS,
    payload: data,
  }
}

export const requestSetLookupCrimeTypes = (data) => {
  return {
    type: REQUEST_SET_LOOKUP_CRIME_TYPES,
    payload: data,
  }
}

export const requestSetLookupNationality = (data) => {
  return {
    type: REQUEST_SET_LOOKUP_NATIONALITY,
    payload: data,
  }
}


export const requestSetUserInfo = (data) => {
  return {
    type: REQUEST_SET_USER_INFO,
    payload: data || {},
  }
}

