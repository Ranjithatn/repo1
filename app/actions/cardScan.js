// @flow
import {
  CARD_SCAN_WORKFLOW,
  CARD_SCAN_RESET,
} from "./actionTypes";

export const cardScanWorkflow = (data) => ({
  type: CARD_SCAN_WORKFLOW,
  payload: data,
});

export const cardScanReset = (data) => ({
  type: CARD_SCAN_RESET,
  payload: data,
});

