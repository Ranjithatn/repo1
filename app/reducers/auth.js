//@flow
import {
  REQUEST_LOGIN,
  REQUEST_LOGIN_SUCCESS,
  REQUEST_LOGIN_FAILED,
  REQUEST_LOGOUT_SUCCESS,
  RECEIVE_INPUT_FIELD_CHANGED,
  REQUEST_SET_NEW_UPDATE,
  REQUEST_SHOW_UPDATE_MESSAGE,
  REQUEST_SET_DOWNLOAD,
  REQUEST_SET_STATUS,
  REQUEST_SET_LOOKUPS,
  REQUEST_SAVE_ANNOTATIONS_CUSTOM,
  REQUEST_SYSTEM_PERMISSION_SUCCESS,
  REQUEST_SET_LOOKUP_CRIME_TYPES,
  REQUEST_SET_LOOKUP_NATIONALITY,
  REQUEST_SET_USER_INFO,
} from "../actions/actionTypes";

const initialState = {
  username: "",
  password: "",
  apiUri: "",
  jwt: "",
  error: "",
  roles: "",
  newUpdate: undefined,
  updateMessage: "",
  newDownload: undefined,
  version: "",
  updateStatus: "",
  parsedRoles: [],
  lookups: [],
  menuPerm:{},
  servicePerm:{}
};

export default function login(state = initialState, action = {}) {
  switch (action.type) {
    case REQUEST_LOGIN:
      return { ...state, error: "" };
    case REQUEST_LOGIN_SUCCESS:
      const { jwt, roles, parsedRoles } = action.payload;
      return { ...state, password: "", jwt, roles, parsedRoles };
    case REQUEST_LOGIN_FAILED:
      return { ...initialState, error: action.payload };
    case REQUEST_LOGOUT_SUCCESS:
      localStorage.removeItem("jwt");
      return initialState;

      return initialState;
    case RECEIVE_INPUT_FIELD_CHANGED:
      if (action.payload.area === "auth") {
        const { field, value } = action.payload;
        let newState = Object.assign({}, state);
        newState[field] = value;
        return newState;
      } else {
        return state;
      }
    case REQUEST_SET_NEW_UPDATE:
      return { ...state, newUpdate: true };
    case REQUEST_SET_STATUS:
      return { ...state, updateStatus: action.payload };
    case REQUEST_SET_DOWNLOAD:
      return { ...state, newDownload: true, version: action.payload };
    case REQUEST_SHOW_UPDATE_MESSAGE:
      return { ...state, updateMessage: action.payload };
    case REQUEST_SET_LOOKUPS:
      return { ...state, lookups: action.payload };
    case REQUEST_SYSTEM_PERMISSION_SUCCESS:
      return {
        ...state,
        // servicePerm: action.payload.service_perm,
        menuPerm: action.payload.menu_perm
      };
    case REQUEST_SET_LOOKUP_CRIME_TYPES:
      return { ...state, lookupCrimeTypes: action.payload };

    case REQUEST_SET_LOOKUP_NATIONALITY:
      return { ...state, nationality: action.payload };

    case REQUEST_SET_USER_INFO:
     return { ...state, userInfo: action.payload };



    default:
      return state;
  }
}
