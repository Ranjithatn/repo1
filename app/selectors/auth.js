import { createSelector } from "reselect";

export const usernameSelector = createSelector(
  state => state.auth.username,
  u => u
);
export const updateMessageSelector = createSelector(
  state => state.auth.updateMessage,
  updateMsg => updateMsg
);
export const downloadSelector = createSelector(
  state => state.auth.newDownload,
  d => d
);
export const updateMsgSelector = createSelector(
  state => state.auth.updateStatus,
  updateStatus => updateStatus
);

export const lookupsSelector = createSelector(
  state => state.auth.lookups,
  lookups => lookups
);



export const userLocationSelector = createSelector(
  state => state.auth.lookups,
  (lookups) => {
    console.log("userLocationSelector::lookups", lookups);
    let result = [];

    if ( lookups.length > 0 ) {
      lookups.find( item => {
        if ( item.lookupName === "LocationIDs" ) {
          result = item.items;
          return true;
        }
      });
    }

    console.log("userLocationSelector::result",result);

    return result;
  }
);

export const lookupCrimeTypesSelector = createSelector(
  state => state.auth.lookupCrimeTypes,
  lookupCrimeTypes => lookupCrimeTypes
);

