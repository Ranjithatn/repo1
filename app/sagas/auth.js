import { call, put, takeLatest, select } from "redux-saga/effects";
import Constants from "../constants";
import * as localStorage from "../utils/localStorage";
import { delay } from "redux-saga";
import { push } from "react-router-redux";
// import { autoUpdater } from "electron-updater";
import {
  REQUEST_LOGIN,
  REQUEST_LOGIN_SUCCESS,
  REQUEST_LOGOUT,
  REQUEST_LOGOUT_SUCCESS,
  REQUEST_SPINNER_START,
  REQUEST_SPINNER_STOP,
  REQUEST_CHECK_FOR_APP_UPDATE,
  REQUEST_LOGIN_POPUP
} from "../actions/actionTypes";
import { requestInputFieldChanged } from '../actions/global';
import {
  requestLoginSuccess,
  requestLogoutSuccess,
  requestLoginFailed,
  requestSetLookups,
  requestSetLookupCrimeTypes,
  requestSetLookupNationality,
  requestSetUserInfo,
} from "../actions/auth";
import { requestShowNotification } from "../actions/notifications";
import {
  // updateServerConnectionStatus,
  requestUpdateServerStatus
} from "../actions/app";
import { updateMsgSelector } from "../selectors/auth";
import { requestActiveJobs, requestSystemPermissions } from "../actions/jobs";
import Api, { url } from "../api/";
import { startSpinner, stopSpinner } from "../../app/actions/spinner";
import { settings } from "../utils/electron";

import { store } from "../index";
import path from "path";
import { translateRes } from "../utils/intl";
import { displayLoginPopup } from "../components/Notification/PortalNotification";

import moment from "moment";


const LivescanAPI = require("../hardwareSDK/biocoreSdk/livescan/livescan");

const electron = require("electron").remote;

const ipcRenderer = require("electron").ipcRenderer;

export function* watchRequestLogin() {
  yield takeLatest(REQUEST_LOGIN, callLogin);
}

export function* watchRequestLoginPopup() {
  yield takeLatest(REQUEST_LOGIN_POPUP, callLoginPopup);
}

export function* watchCheckForappUpdate() {
  yield takeLatest(REQUEST_CHECK_FOR_APP_UPDATE, checkForAppUpdate);
}

export function* watchRequestLogout() {
  yield takeLatest(REQUEST_LOGOUT, callLogout);
}

export function* checkForAppUpdate(action) {
  const _settings = settings();

  let newApiUri = "";
  if (_settings && _settings.ipAddress) {
    newApiUri = `${_settings.protocol ? _settings.protocol : "http"}://${
      _settings.ipAddress
    }/services-gateway-ucw/api`;
    Constants.API_ORIGIN_URL = newApiUri;
    localStorage.set("api", newApiUri);
  }

  yield put(startSpinner());
  yield call(delay, 1000);
  ipcRenderer.send("checkForUpdates");
}

export function* callLogin(action) {
  yield put(stopSpinner());
  try {
    yield put(startSpinner());
    const { payload } = action;
    const _settings = settings();

    let newApiUri = "";
    if (_settings && _settings.ipAddress) {
      newApiUri = `${_settings.protocol ? _settings.protocol : "http"}://${
        _settings.ipAddress
      }/services-gateway-ucw/api`;
      Constants.API_ORIGIN_URL = newApiUri;
      localStorage.set("api", newApiUri);
    }
    // localStorage.set("api", Constants.API_ORIGIN_URL);

    const loginRes = yield call(Api, {
      url: url.auth.login,
      method: "POST",
      data: payload
    });
    if (loginRes.jwt) {
      const { jwt } = loginRes;
      // const {
      //   "http://schemas.microsoft.com/ws/2008/06/identity/claims/role": roles
      // } = yield call(parseJwt, jwt);

      const parsed = yield call(parseJwt, jwt);

      // console.log( "userRoles", parsed.userRoles );
      // console.log( "userRoles::JSON", JSON.parse(parsed.userRoles) );
      // const parsedRoles = JSON.parse(parsed.userRoles);
      const parsedRoles = parsed.userRoles;

      let {
        "http://schemas.microsoft.com/ws/2008/06/identity/claims/role": roles
      } = parsed;

      // console.log("AUTH::roles",roles);
      if (!roles) {
        roles = "custom";
      }

      yield call(localStorage.set, "jwt", jwt);

      yield put(requestLoginSuccess({ jwt, roles, parsedRoles }));
      yield put( requestSystemPermissions() );
      yield call( requestLookupCrimeTypes );
      yield call( getUserLocations );
      // yield call( requestLookupNationality );

      yield put( requestSetUserInfo(loginRes.userInfo) );

      yield put(requestActiveJobs());
      yield put(push("/authenticated/jobqueue"));

      yield put(stopSpinner());

      if (localStorage.get("loginpopup")) {
        localStorage.remove("loginpopup");
      }

      yield call(startCheckingServerConnectionStatus);
      yield call(checkForTokenExpiration);
    } else {
      if (loginRes.Error) {
        yield put(requestLoginFailed(loginRes.Error));
      }
      yield put(stopSpinner());
      yield put(requestInputFieldChanged( { target: { id: 'username', value: '', } }, 'auth'));
      yield put(requestInputFieldChanged( { target: { id: 'password', value: '', } }, 'auth'));
    }
  } catch (e) {
    yield put(requestLoginFailed("Login failed for reasons unknown"));
  } finally {
    yield put(stopSpinner());
    // yield call(startCheckingServerConnectionStatus);
  }
}

export function* callLoginPopup(action) {
  try {
    localStorage.set("loginpopup", "true");
    yield put(startSpinner());
    const { payload } = action;
    const _settings = settings();

    let newApiUri = "";
    if (_settings && _settings.ipAddress) {
      newApiUri = `${_settings.protocol ? _settings.protocol : "http"}://${
        _settings.ipAddress
      }/services-gateway-ucw/api`;
      Constants.API_ORIGIN_URL = newApiUri;
      localStorage.set("api", newApiUri);
    }
    localStorage.set("api", Constants.API_ORIGIN_URL);

    const loginRes = yield call(Api, {
      url: url.auth.login,
      method: "POST",
      data: payload
    });

    if (loginRes.jwt) {
      const { jwt } = loginRes;

      const parsed = yield call(parseJwt, jwt);

      // const parsedRoles = JSON.parse(parsed.userRoles);
      const parsedRoles = parsed.userRoles;

      let {
        "http://schemas.microsoft.com/ws/2008/06/identity/claims/role": roles
      } = parsed;

      if (!roles) {
        roles = "custom";
      }

      yield call(localStorage.set, "jwt", jwt);

      yield put(requestLoginSuccess({ jwt, roles, parsedRoles }));
      yield put(requestSystemPermissions())

      yield call( requestLookupCrimeTypes );
      yield call( getUserLocations );
      // yield call( requestLookupNationality );

      yield put( requestSetUserInfo(loginRes.userInfo) );

      yield put(requestActiveJobs());
      // yield put(push("/authenticated/jobqueue"));
      localStorage.remove("loginpopup");

      yield put(stopSpinner());
    } else {
      if (loginRes.props.message) {
        // displayLoginPopup( loginRes.props.message );
      }
      yield put(stopSpinner());
    }
  } catch (e) {
    displayLoginPopup("Login failed for reasons unknown");
  } finally {
    yield put(stopSpinner());
  }
}

export function* callLogout(action) {
  yield put(startSpinner());
  try {
    const { history } = action;
    const logoutRes = yield call(Api, {
      url: url.auth.logout,
      method: "POST",
      isJwtRequired: true,
      data: {}
    });
    if (logoutRes) {
      if (logoutRes.message) {
        yield put(
          requestShowNotification({
            message: translateRes(logoutRes.message),
            type: "is-success"
          })
        );
      }

      // stop livescan
      LivescanAPI.stopCapture();
      // stop cardscan
      require("dwt");

      Dynamsoft.WebTwainEnv.Unload();

      yield call(localStorage.set, "jwt", "");
      yield put(requestLogoutSuccess());
      yield put(push("/"));
    }
  } catch (e) {
    console.error("-------- logout failed", e);
  } finally {
    yield put(stopSpinner());
  }
}

export function parseJwt(token) {
  var base64Url = token && token.split(".")[1];
  var base64 = base64Url.replace("-", "+").replace("_", "/");

  const parsed = JSON.parse(window.atob(base64));

  return parsed;
}

export function* handleConnectionStatus() {
  const connStatus = yield call(Api, {
    url: url.systeminfo.systeminfo,
    method: "GET",
    isJwtRequired: true,
    data: {}
  });

  if (connStatus) {
    // console.log("connectionStatus API response", connStatus);
    if (connStatus.serverLiveStatus && connStatus.serverLiveStatus.length > 0) {
      let status = "unknown";
      if (connStatus.serverLiveStatus[0].isActive) {
        status = "connected";
      } else {
        status = "disconnected";
      }
      yield put(requestUpdateServerStatus(status));
    }
  } // if ends here
}

export function* startCheckingServerConnectionStatus() {
  const _settings = settings();
  let callTime = 0;

  if (_settings.serverConnection && _settings.serverConnection > 0) {
    callTime = _settings.serverConnection;
  }

  if (callTime > 0) {
    while (localStorage.get("jwt") && localStorage.get("jwt") != "") {
      // console.log("inside while loop");
      // user is logged in lets continue
      yield call(handleConnectionStatus);
      yield call(delay, callTime * 1000);
    }
  } else {
    // console.log("server connection time is not specified");
    yield call(handleConnectionStatus);
  }
}

export function* getUserLocations() {
  // console.log("getUserLocations::init");

  try {
    const apiCall = yield call(Api, {
      url: url.systeminfo.lookups,
      method: "GET",
      isJwtRequired: true,
      data: {}
    });

    if (apiCall) {
      // console.log("apiCall::success", apiCall);
      yield put(requestSetLookups(apiCall));
    } else {
      console.log("apiCall::failed", apiCall);
    }
  } catch (e) {
    console.log("getUserLocations::catch::error occoured", e);
  }
}


export function* requestLookupCrimeTypes() {
  try {
    const apiCall = yield call(Api, {
      url: url.systeminfo.getniclookupcrimetypes,
      method: "GET",
      isJwtRequired: true,
      data: {}
    });

    if (apiCall) {
      // console.log("requestLookupCrimeTypes::apiCall::success", apiCall);
      yield put( requestSetLookupCrimeTypes(apiCall) );
    } else {
      // console.log("requestLookupCrimeTypes::apiCall::failed", apiCall);
    }
  } catch (e) {
    console.log("requestLookupCrimeTypes::catch::error occoured", e);
  }

}




export function* requestLookupNationality() {
  try {
    const apiCall = yield call(Api, {
      url: url.systeminfo.nationality,
      method: "GET",
      isJwtRequired: true,
      data: {},
      fromBase: true,
    });

    if (apiCall) {
      yield put( requestSetLookupNationality(apiCall) );
    }
  } catch (e) {
    console.log("requestLookupNationality::catch::error occoured", e);
  }

}






export function* handleTokenExpiration() {
  console.log("handleTokenExpiration");

  try {

    const jwt = localStorage.get("jwt");

    if ( jwt ) {
      const parsed = parseJwt(jwt);
      const expiryDate = new Date( parsed.exp * 1000 );
      const currentDate = new Date();

      let a = moment(expiryDate);
      let b = moment(currentDate);
      let minsLeft = a.diff(b, 'minutes');
      console.log("minutes left until token expiry",minsLeft);


      if ( minsLeft <= 1 ) {
            
        const newToken = yield call(Api, {
          url: url.auth.refreshToken,
          method: "POST",
          isJwtRequired: true,
          data: {}
        });

        console.log("newToken",newToken);
        if ( newToken ) {
          // localStorage.set("jwt", newToken);
          yield call(localStorage.set, "jwt", newToken.jwt);
        }

      }

    }

  } catch (e) {
    console.log("handleTokenExpiration::catch",e);
  }


}



export function* checkForTokenExpiration() {
  let callTime = 60000; // in ms

  while (localStorage.get("jwt") && localStorage.get("jwt") != "") {
    yield call(handleTokenExpiration);
    yield call(delay, callTime);
  }

}