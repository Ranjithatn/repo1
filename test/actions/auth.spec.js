import { spy } from "sinon";
import * as actions from "../../app/actions/auth";
import {
  REQUEST_LOGIN,
  REQUEST_LOGOUT,
  REQUEST_LOGIN_SUCCESS,
  REQUEST_LOGOUT_SUCCESS,
  REQUEST_SET_NEW_UPDATE,
  REQUEST_SHOW_UPDATE_MESSAGE,
  REQUEST_SET_DOWNLOAD,
  REQUEST_CHECK_FOR_APP_UPDATE
} from "../../app/actions/actionTypes";

const history = {
  push: spy()
};

describe("actions", () => {
  let output;
  let expected;
  it("should create login action", () => {
    const payload = {
      username: "username",
      password: "password"
    };
    output = actions.requestLogin(payload, history);
    expected = {
      type: REQUEST_LOGIN,
      payload: {
        username: "username",
        password: "password"
      },
      history
    };
    expect(output).toEqual(expected);
    expect(output).toMatchSnapshot();
  });
  it("should create login success action", () => {
    const authData = {
      jwt: "sdfsdergdgdhddh",
      roles: "admin"
    };
    output = actions.requestLoginSuccess(authData);
    expected = {
      type: REQUEST_LOGIN_SUCCESS,
      payload: authData
    };
    expect(output).toEqual(expected);
    expect(output).toMatchSnapshot();
  });

  it("should create logout action", () => {
    output = actions.requestLogout();
    expected = {
      type: REQUEST_LOGOUT
    };
    expect(output).toEqual(expected);

    output = actions.requestLogout(history);
    expect(output).toMatchSnapshot();
  });
  it("should create logout success action", () => {
    output = actions.requestLogoutSuccess();
    expected = {
      type: REQUEST_LOGOUT_SUCCESS
    };
    expect(output).toEqual(expected);

    output = actions.requestLogoutSuccess();
    expect(output).toMatchSnapshot();
  });
  it("should create requestSetNewUpdate action", () => {
    output = actions.requestSetNewUpdate();
    expected = {
      type: REQUEST_SET_NEW_UPDATE
    };
    expect(output).toEqual(expected);

    output = actions.requestSetNewUpdate();
    expect(output).toMatchSnapshot();
  });
  it("should create requestSetDownload action", () => {
    output = actions.requestSetDownload("1.0.1");
    expected = {
      type: REQUEST_SET_DOWNLOAD,
      payload:"1.0.1"
    };
    expect(output).toEqual(expected);

    output = actions.requestSetDownload("1.0.1");
    expect(output).toMatchSnapshot();
  });
  it("should create requestShowUpdateMessage action", () => {
    output = actions.requestShowUpdateMessage("1.0.1");
    expected = {
      type: REQUEST_SHOW_UPDATE_MESSAGE,
      payload:"1.0.1"
    };
    expect(output).toEqual(expected);

    output = actions.requestShowUpdateMessage("1.0.1");
    expect(output).toMatchSnapshot();
  });
  it("should create REQUEST_CHECK_FOR_APP_UPDATE action", () => {
    output = actions.requestCheckForAppUpdate();
    expected = {
      type: REQUEST_CHECK_FOR_APP_UPDATE
    };
    expect(output).toEqual(expected);

    output = actions.requestShowUpdateMessage();
    expect(output).toMatchSnapshot();
  });
});
