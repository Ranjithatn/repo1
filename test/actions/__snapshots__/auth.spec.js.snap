// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should create REQUEST_CHECK_FOR_APP_UPDATE action 1`] = `
Object {
  "payload": undefined,
  "type": "REQUEST_SHOW_UPDATE_MESSAGE",
}
`;

exports[`actions should create login action 1`] = `
Object {
  "history": Object {
    "push": [Function],
  },
  "payload": Object {
    "password": "password",
    "username": "username",
  },
  "type": "REQUEST_LOGIN",
}
`;

exports[`actions should create login success action 1`] = `
Object {
  "payload": Object {
    "jwt": "sdfsdergdgdhddh",
    "roles": "admin",
  },
  "type": "REQUEST_LOGIN_SUCCESS",
}
`;

exports[`actions should create logout action 1`] = `
Object {
  "history": Object {
    "push": [Function],
  },
  "type": "REQUEST_LOGOUT",
}
`;

exports[`actions should create logout success action 1`] = `
Object {
  "type": "REQUEST_LOGOUT_SUCCESS",
}
`;

exports[`actions should create requestSetDownload action 1`] = `
Object {
  "payload": "1.0.1",
  "type": "REQUEST_SET_DOWNLOAD",
}
`;

exports[`actions should create requestSetNewUpdate action 1`] = `
Object {
  "type": "REQUEST_SET_NEW_UPDATE",
}
`;

exports[`actions should create requestShowUpdateMessage action 1`] = `
Object {
  "payload": "1.0.1",
  "type": "REQUEST_SHOW_UPDATE_MESSAGE",
}
`;
